Tests/MaterialLib/TestCapillaryPressureSaturationModel.cpp

```diff
@@ -0,0 +1,101 @@
+/**
+ * \copyright
+ * Copyright (c) 2012-2016, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ * \file   TestCapillaryPressureSaturationModel.cpp
+ *
+ * Created on November 1, 2016, 11:06 AM
+ */
+
+#include <gtest/gtest.h>
+
+#include <memory>
+#include <vector>
+
+#include "BaseLib/ConfigTree.h"
+
+#include "TestTools.h"
+
+#include "MaterialLib/PorousMedium/UnsaturatedProperty/CapillaryPressure/CapillaryPressureSaturation.h"
+#include "MaterialLib/PorousMedium/UnsaturatedProperty/CapillaryPressure/CreateCapillaryPressureModel.h"
+
+using namespace MaterialLib;
+using namespace MaterialLib::PorousMedium;
+
+std::unique_ptr<CapillaryPressureSaturation> createCapillaryPressureModel(
+    const char xml[])
+{
+    auto const ptree = readXml(xml);
+    BaseLib::ConfigTree conf(ptree, "", BaseLib::ConfigTree::onerror,
+                             BaseLib::ConfigTree::onwarning);
+    auto const& sub_config = conf.getConfigSubtree("capillary_pressure");
+    return MaterialLib::PorousMedium::createCapillaryPressureModel(sub_config);
+}
+
+TEST(MaterialPorousMedium, checkBrookCoreyCapillaryPressure)
+{
+    const char xml[] =
+        "<capillary_pressure>"
+        "   <type>BrookCorey</type>"
+        "   <pd> 19600.0 </pd> "
+        "   <sr> 0.1 </sr> "
+        "   <smax> 1. </smax> "
+        "   <m> 2 </m> "
+        "   <pc_max> 1.e11 </pc_max> "
+        "</capillary_pressure>";
+    auto const pc_model = createCapillaryPressureModel(xml);
+
+    std::vector<double> S = {0.11, 0.2, 0.3, 0.44, 0.52, 0.6, 0.85};
+    // The following expected data are calculated by using  OGS5.
+    std::vector<double> pc = {185941.926417901, 58800,
+                              41577.878733769,  31888.7772993424,
+                              28691.4621446869, 26296.1594153975,
+                              21470.7242542025};
+    std::vector<double> dpc_dS = {-9297096.32089506, -294000,
+                                  -103944.696834422, -46895.2607343271,
+                                  -34156.5025531986, -26296.1594153975,
+                                  -14313.8161694683};
+
+    for (std::size_t i = 0; i < S.size(); i++)
+    {
+        ASSERT_NEAR(pc[i], pc_model->getCapillaryPressure(S[i]), 1.e-5);
+        ASSERT_NEAR(S[i], pc_model->getSaturation(pc[i]), 1.e-5);
```

Comment:
That doesn't seem too strict, since S is on the order of 0.1...1
Or is it impossible to decrease this tolerance.